src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -300,73 +300,80 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
-          toggleShowMore: this.cachedToggleShowMore,
+          ...this.getWidgetRenderState(initOptions),
+          instantSearchInstance: initOptions.instantSearchInstance,
         });
       },
 
       render(renderOptions) {
-        const {
-          results,
-          state,
-          createURL,
-          instantSearchInstance,
-        } = renderOptions;
-
-        const facetValues = results.getFacetValues(attribute, { sortBy }) || [];
-        const items = transformItems(
-          facetValues.slice(0, this.getLimit()).map(formatItems)
-        );
-
-        const maxValuesPerFacetConfig = state.maxValuesPerFacet;
-        const currentLimit = this.getLimit();
-        // If the limit is the max number of facet retrieved it is impossible to know
-        // if the facets are exhaustive. The only moment we are sure it is exhaustive
-        // is when it is strictly under the number requested unless we know that another
-        // widget has requested more values (maxValuesPerFacet > getLimit()).
-        // Because this is used for making the search of facets unable or not, it is important
-        // to be conservative here.
-        hasExhaustiveItems =
-          maxValuesPerFacetConfig > currentLimit
-            ? facetValues.length <= currentLimit
-            : facetValues.length < currentLimit;
+        render({
+          ...this.getWidgetRenderState(renderOptions),
+          instantSearchInstance: renderOptions.instantSearchInstance,
+        });
+      },
 
-        lastResultsFromMainSearch = items;
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          refinementList: {
+            ...renderState.refinementList,
+            [attribute]: this.getWidgetRenderState(renderOptions),
+          },
+        };
+      },
 
-        this.toggleShowMore = this.createToggleShowMore(renderOptions);
+      getWidgetRenderState(renderOptions) {
+        const { results, state, createURL } = renderOptions;
+        const isFirstSearch = !results;
+        let items = [];
+
+        if (results) {
+          const facetValues =
+            results.getFacetValues(attribute, { sortBy }) || [];
+          items = transformItems(
+            facetValues.slice(0, this.getLimit()).map(formatItems)
+          );
+
+          const maxValuesPerFacetConfig = state.maxValuesPerFacet;
+          const currentLimit = this.getLimit();
+          // If the limit is the max number of facet retrieved it is impossible to know
+          // if the facets are exhaustive. The only moment we are sure it is exhaustive
+          // is when it is strictly under the number requested unless we know that another
+          // widget has requested more values (maxValuesPerFacet > getLimit()).
+          // Because this is used for making the search of facets unable or not, it is important
+          // to be conservative here.
+          hasExhaustiveItems =
```

Comment:
where's this variable used now?

Reply:
looks like L210 and L193, but I think it might be better if we put this in a function or something? Would be a bit more recalculating, but not sure if that's a big deal

Reply:
it's also passed to renderFn in showMore, but not in getWidgetRenderState. We should make that consistent and expose it here + use getWidgetRenderState in the renderFn in showMore